addons/event/models/event_registration.py

```diff
@@ -83,25 +98,34 @@ def _prepare_attendee_values(self, registration):
         data.update({key: value for key, value in registration.items() if key in self._fields})
         return data
 
+    def name_get(self):
+        ret_list = []
+        for registration in self:
+            if not registration.partner_id or not registration.partner_id.name:
+                name = registration.name
+            elif not registration.name or registration.name == registration.partner_id.name:
+                name = registration.partner_id.name
+            else:
+                name = registration.partner_id.name + ', ' + registration.name
+            ret_list.append((registration.id, name))
+        return ret_list
+
     def do_draft(self):
         self.write({'state': 'draft'})
 
     def confirm_registration(self):
```

Comment:
We could rename confirm_registration to action_confirm so that everything is well named :) .